packages/components/src/hooks/useDgsData/fetchDataFromDgsApi.ts

```diff
@@ -0,0 +1,61 @@
+import type {
+  DgsApiDatasetSearchParams,
+  DgsApiDatasetSearchResponse,
+  DgsApiDatasetSearchResponseSuccess,
+} from "./types"
+
+export const fetchDataFromDgsApiDataset = async (
+  params: DgsApiDatasetSearchParams,
+): Promise<DgsApiDatasetSearchResponseSuccess> => {
+  const url = generateDgsUrl(params)
+  const res = await fetch(url)
+
+  if (!res.ok) {
+    throw new Error("Failed to fetch data from DGS API")
+  }
+
+  const data = (await res.json()) as DgsApiDatasetSearchResponse
```

Comment:
nit: not really best practice but can we create a DD synthetic monitor to make sure the shape of the response from DGS has not changed?